tests/php/Dev/DeprecationTest.php

```diff
@@ -8,103 +8,33 @@
-        Deprecation::notification_version('2.0.0-beta1');
-        $this->assertNull(Deprecation::notice('2.0', 'Deprecation test failed'));
-        $this->assertNull(Deprecation::notice('2.0.0', 'Deprecation test failed'));
-    }
-
-    public function testGreaterVersionTriggersNotice()
-    {
-        $this->expectError();
-        Deprecation::notification_version('3.0.0');
-        Deprecation::notice('2.0', 'Deprecation test passed');
-    }
-
-    public function testNonMatchingModuleNotifcationVersionDoesntAffectNotice()
-    {
-        Deprecation::notification_version('1.0.0');
-        Deprecation::notification_version('3.0.0', 'my-unrelated-module');
-        $this->callThatOriginatesFromFramework();
-    }
-
-    public function testMatchingModuleNotifcationVersionAffectsNotice()
-    {
-        $this->expectError();
-        Deprecation::notification_version('1.0.0');
-        Deprecation::notification_version('3.0.0', 'silverstripe/framework');
-        $this->callThatOriginatesFromFramework();
-    }
-
-    public function testMethodNameCalculation()
-    {
-        $this->assertEquals(
-            TestDeprecation::get_method(),
-            static::class . '->testMethodNameCalculation'
-        );
-    }
-
-    public function testScopeMethod()
-    {
-        $this->expectError();
-        $this->expectErrorMessage('DeprecationTest->testScopeMethod is deprecated. Method scope');
-        Deprecation::notification_version('2.0.0');
-        Deprecation::notice('2.0.0', 'Method scope', Deprecation::SCOPE_METHOD);
-    }
-
-    public function testScopeClass()
-    {
-        $this->expectError();
-        $this->expectErrorMessage('DeprecationTest is deprecated. Class scope');
-        Deprecation::notification_version('2.0.0');
-        Deprecation::notice('2.0.0', 'Class scope', Deprecation::SCOPE_CLASS);
-    }
-
-    public function testScopeGlobal()
-    {
-        $this->expectError();
-        $this->expectErrorMessage('Global scope');
-        Deprecation::notification_version('2.0.0');
-        Deprecation::notice('2.0.0', 'Global scope', Deprecation::SCOPE_GLOBAL);
+        $this->expectErrorMessage($message);
+        Deprecation::$notice_level = E_USER_NOTICE;
+        Deprecation::enable();
+        Deprecation::notice('1.2.3', 'My message');
     }
 
-    protected function callThatOriginatesFromFramework()
+    public function testDisabled()
     {
-        $module = TestDeprecation::get_module();
-        $this->assertNotNull($module);
-        $this->assertEquals('silverstripe/framework', $module->getName());
-        $this->assertNull(Deprecation::notice('2.0', 'Deprecation test passed'));
+        Deprecation::$notice_level = E_USER_NOTICE;
+        // test that no error error is raised because by default Deprecation is disabled
+        Deprecation::notice('4.5.6', 'My message');
+        // this is done so that phpunit does not complain about a test with no assertions
+        $this->assertTrue(true);
```

Comment:
I'd prefer add the `@doesNotPerformAssertions` annotation to the method's PHPDoc instead of asserting true - but given that asserting true is true has been used a lot already I'm not fussed if you choose to not make that change.

Reply:
Updated